Refined settings overhaul

YodaForce157 opened the pull request:
Fixed the prior PR.
**Added**:
- Decorations
- Settings
- Codecs
- SettingsHandler
- Create treeview
- settings to & from json
- Remove settings from confighandler

**Latest TODO**
- [x] Seperate settings tab in half, left side for the folder tree, and right site for the settings and parent folder tree

~~**TODO**~~
~~- [x] When settings are **not** applied, on exit of settings window, load settings from file a.k.a reset back to old settings.~~
~~- [x] Fix methods for trees/to/fromjson~~
~~- [x] Localize settings (Might do this at a later point in time)~~
~~- [x] Make it look, *better*~~

~~**Currently Broken**~~
~~- Github plugin~~
~~- Plugin settings~~
~~- Update theme method in Railroad.class~~

xf8b commented on README.md:
```diff
@@ -26,6 +26,10 @@ There is no website available.
 
+### How do I help? 
+
+The first thing I would recomened is to join the discord server, you can talk with other contributers to railroad about contributing yourself. I would then (of course) make your own fork and look at what needs to be done, Then just make a pull request, someone will review it. And thats all to contributing to Railroad.
```
this is fine except for the spelling and grammer but it is ok

xf8b commented on README.md:
```diff
@@ -28,7 +28,7 @@ To report a bug go [here](https://github.com/Railroad-Team/Railroad/issues). Use
 ### How do I help? 
 
-The first thing I would recomened is to join the discord server, you can talk with other contributers to railroad about contributing yourself. I would then (of course) make your own fork and look at what needs to be done, Then just make a pull request, someone will review it. And thats all to contributing to Railroad.
+The first thing that is recomeneded is to join the discord server, you can talk with others who contribute to railroad about contributing yourself. Then (of course) make your own fork and look at what needs to be done, Then just make a pull request, someone will review it. And thats all to contributing to Railroad.
```
using a list is probably more cleaner, so maybe something like this

```diff
- ### How do I help?
+ ### How do I contribute?

- The first thing that is recomeneded is to join the discord server, you can talk with others who contribute to railroad about contributing yourself. Then (of course) make your own fork and look at what needs to be done, Then just make a pull request, someone will review it. And thats all to contributing to Railroad.
+ 1. **(recommended)** Join the Discord server, which allows you to talk and discuss with other contributors.
+ 2. Fork the repository and make your changes.
+ 3. Make a pull request to this repository.

KingRealzYT commented:
Hmm yea, That does indeed look way cleaner.

KingRealzYT commented:
I put recommended in caps to make it show like (important)

xf8b commented:
you remove one m

KingRealzYT commented:
just fixed it

Commit:
Moved across system to new branch, also cleaned up code an created an ID system dedicated to codecs

Commit:
Fix createTree method, create decorations, remove unnecessary files and methods.

Need to update the Github Plugin for it to work with these new changes. Therefore, it is currently disabled. (It would be anyways due to it needing some weird config file change + it used the old settings pane gui)

Commit:
Remove javadoc downloader I comitted by accident, and optimize imports

YodaForce157 commented on src/main/java/io/github/railroad/Railroad.java:
```diff
@@ -171,6 +170,10 @@ public void start(Stage primaryStage) {
             HTTP_CLIENT.dispatcher().executorService().shutdown();
             HTTP_CLIENT.connectionPool().evictAll();
         });
+        } catch (Exception e) {
```
Possibly keep this here, otherwise it may unexpectedly close on the end user, and this could be used for simple debugging?

Commit:
Another unused enum, Remove pointless code.

Commit:
Localize settings tree

DaRealTurtyWurty committed:
Fixed generics(ish)

Commit:
Fix fromJson and minor theme changes

Commit:
attempt to make it look better

YodaForce157 commented:
Ready for merge I believe. Will need to work out the plugin system & plugin settings system at a later point.

Commit:
Merge branch 'dev' into RefinedSettingsOverhaul

YodaForce157 commented:
Before
![image](https://github.com/user-attachments/assets/8bad5481-55a9-4866-acd6-92f4273dadf3)

After
![image](https://github.com/user-attachments/assets/1938d5d3-74a6-4587-8049-94cb003e3f70)

wolfieboy09 commented:
Are the images in the right order?

YodaForce157 commented:
Yes.

wolfieboy09 commented:
I perfered the "before" part with the images on the side, and that such thing
![image](https://github.com/user-attachments/assets/d294a78d-71be-4398-8130-6290a2fa9dca)

Commit:
Separate folders and settings

Commit:
Create basic search functionality - Needs polishing

Commit:
Fully functional, needs a good bit of GUI work though. Also need to clean some stuff up

Commit:
Search works! GUI looks better. Need to add a few more minor tweaks and then clean up code.

Commit:
Use lang files to load search strings, add todos too.

Commit:
Search fully (to my knowledge) works, missing a few improvements but fully functional.

Commit:
Make the settings actually work, Forgot to add the event handlers to the settings nodes. Also fix the contents pane not working when localized

Commit:
Implement FuzzySearch

Commit:
Clean code

Commit:
Move language to Railroad>General, display the current path, and a few minor UI & lang changes

Commit:
Add margin to description label

DaRealTurtyWurty committed:
Update Indexes.java
